packages/near-lake-primitives/src/types/block.ts

```diff
@@ -0,0 +1,159 @@
+import { Action, isActionReceipt, outcomeWithReceiptToReceipt, receiptViewToAction } from './receipts';
+import { StreamerMessage, ValidatorStakeView } from './core/types';
+import { Receipt } from './receipts';
+import { Transaction } from './transactions';
+import { Event, logToRawEvent } from './events';
+import { fromStateChangeViewToStateChange, StateChange } from './stateChanges';
+
+export class Block {
+    readonly streamerMessage: StreamerMessage;
+    private executedReceipts: Receipt[];
+    readonly postponedReceipts: Receipt[];
+    readonly transactions: Transaction[];
+    private _actions: Map<string, Action>;
```

Comment:
Just curious, why `Map` instead of just an Object?

Reply:
`Map` has more features such as keeping track of insertion order to make it easily iteratable which we might want to do for _actions as well as it keeps track of the size of the hash map which we would have to manually keep track of if we used an Object. Though, a `Map` requires more memory.